stages/org.osbuild.ostree.deploy

```diff
@@ -104,44 +131,93 @@ def make_fs_identifier(desc):
     raise ValueError("unknown rootfs type")
 
 
-def main(tree, inputs, options):
-    osname = options["osname"]
-    rootfs = options.get("rootfs")
-    mounts = options.get("mounts", [])
-    kopts = options.get("kernel_opts", [])
-    ref = options["ref"]
-    remote = options.get("remote")
+def ostree_commit_deploy(tree, inputs, osname, remote, ref, kopts):
+    if len(inputs) == 0:
+        if not ref:
+            raise ValueError("ref should be specified in options")
+    elif len(inputs) == 1:
+        if ref:
+            raise ValueError("Should not specify ref if input was specified")
 
-    # If provided an input then do the pull into the tree
-    if len(inputs) != 0:
+        # If we have an input then we need to pull_local() from the input
+        # first before we deploy.
         source_repo, commits = ostree.parse_input_commits(inputs["commits"])
         target_repo = f"{tree}/ostree/repo"
         for commit, data in commits.items():
-            loopref = data.get("ref", commit)
-            ostree.pull_local(source_repo, target_repo, remote, loopref)
+            ref = data.get("ref", commit)
+            ostree.pull_local(source_repo, target_repo, remote, ref)
 
     if remote:
         ref = f"{remote}:{ref}"
 
-    kargs = []
+    kargs = [f'--karg-append={v}' for v in kopts]
+    ostree.cli("admin", "deploy", ref,
+               *kargs, sysroot=tree, os=osname)
+
+
+def ostree_container_deploy(tree, inputs, osname, remote, target_gpgcheck, kopts):
+    images = containers.parse_containers_input(inputs)
+    for image in images.values():
+        with containers.container_source(image) as (image_name, image_source):
+            extra_args = []
+            imgref = f"ostree-unverified-image:{image_source}"
+
+            if target_gpgcheck:
+                target_imgref = 'ostree-remote-registry'
+            else:
+                target_imgref = 'ostree-unverified-registry'
+
+            if remote:
+                # If the user specified an ostree remote it's most likely
+                # because they want to take advantage of GPG signature
+                # verification of the OSTree commits inside the image.
+                # Add it to the imgref spec here.
+                target_imgref += f':{remote}'
```

Comment:
Mmm...this isn't quite right.  If `remote` is specified then we should key off of it unconditionally and generate e.g. `target_imgref = f'ostree-remote-registry:{remote}:{image_name}` and be done.

Something more like:

```
if remote:
    target_imgref = f'ostree-remote-registry:{remote}:{image_name}
elif target_signature_verification:
    target_imgref = f'ostree-image-signed:registry:{image_name}'
else:
    target_imgref = f'ostree-unverified-registry:{image_name}'
```

I think.

(And yes we do need a shorthand for `ostree-image-signed:registry`...but we need to make that stink less...)

Reply:
ok I think the missing piece of the puzzle for me was I didn't fully understand we needed to specify: `ostree-image-signed:registry`. Makes sense now.